test/python/dagcircuit/test_dagcircuit.py

```diff
@@ -2131,5 +2132,97 @@ def test_clbit_conditional(self):
+        dag = circuit_to_dag(circuit)
+        op_nodes = dag.op_nodes()
+        dag.swap_nodes(op_nodes[0], op_nodes[1])
+        swapped_circuit = dag_to_circuit(dag)
+
+        expected_circuit = QuantumCircuit(qreg)
+        expected_circuit.rx(pi / 2, qreg[0])
+        expected_circuit.rx(pi, qreg[0])
+        self.assertEqual(swapped_circuit, expected_circuit)
+
+    def test_2q_swap_fully_connected(self):
+        """test swaping full connected 2q gates"""
+        from qiskit.converters import dag_to_circuit
+
+        circuit = QuantumCircuit(2)
+        circuit.cx(0, 1)
+        circuit.cy(1, 0)
+        dag = circuit_to_dag(circuit)
+        op_nodes = dag.op_nodes()
+        dag.swap_nodes(op_nodes[0], op_nodes[1])
+        swapped_circuit = dag_to_circuit(dag)
+
+        expected_circuit = QuantumCircuit(2)
+        expected_circuit.cy(1, 0)
+        expected_circuit.cx(0, 1)
+        self.assertEqual(swapped_circuit, expected_circuit)
+
+    def test_2q_swap_partially_connected(self):
+        """test swapping 2q partially connected gates"""
+        from qiskit.converters import dag_to_circuit
+
+        circuit = QuantumCircuit(3)
+        circuit.cx(1, 0)
+        circuit.cz(1, 2)
+        dag = circuit_to_dag(circuit)
+        op_nodes = dag.op_nodes()
+        dag.swap_nodes(op_nodes[0], op_nodes[1])
+        swapped_circuit = dag_to_circuit(dag)
+
+        expected_circuit = QuantumCircuit(3)
+        expected_circuit.cz(1, 2)
+        expected_circuit.cx(1, 0)
+        self.assertEqual(swapped_circuit, expected_circuit)
+
+    def test_4q_swap_partially_connected(self):
+        """test swapping 4q partially connected gates"""
+        from qiskit.converters import dag_to_circuit
+        from qiskit.circuit.library.standard_gates import C3XGate
+
+        qreg = QuantumRegister(5)
+        circuit = QuantumCircuit(qreg)
+        circuit.append(C3XGate(), qreg[[0, 1, 2, 3]])
+        circuit.append(C3XGate(), qreg[[0, 1, 2, 4]])
+        dag = circuit_to_dag(circuit)
+        op_nodes = dag.op_nodes()
+        dag.swap_nodes(op_nodes[0], op_nodes[1])
+        swapped_circuit = dag_to_circuit(dag)
+
+        expected_circuit = QuantumCircuit(qreg)
+        expected_circuit.append(C3XGate(), qreg[[0, 1, 2, 4]])
+        expected_circuit.append(C3XGate(), qreg[[0, 1, 2, 3]])
+        self.assertEqual(swapped_circuit, expected_circuit)
+
+    def test_2q_swap_non_connected(self):
+        """test swapping 2q non-connected gates"""
+        from qiskit.converters import dag_to_circuit
+
+        circuit = QuantumCircuit(4)
+        circuit.cx(1, 0)
+        circuit.cz(2, 3)
+        dag = circuit_to_dag(circuit)
+        op_nodes = dag.op_nodes()
+        dag.swap_nodes(op_nodes[0], op_nodes[1])
+        swapped_circuit = dag_to_circuit(dag)
+
+        expected_circuit = QuantumCircuit(4)
+        expected_circuit.cx(1, 0)
+        expected_circuit.cz(2, 3)
+        self.assertEqual(swapped_circuit, expected_circuit)
+
```

Comment:
These tests seem fairly conclusive for the case of only having the two operations in question in the circuit.  Can we add some tests that have operation successors and predecessors of the two nodes as well, potentially with the successors/predecessors having unrelated edges connecting them to one of the pair?

I imagine it should work, since we're already dealing with the in- and out-nodes, but the multi-wire case would be good.

Reply:
I couple of tests of this nature have been added